source/renderer/app/stores/WalletsStore.js

```diff
@@ -869,21 +875,43 @@ export default class WalletsStore extends Store {
     });
   };
 
-  isValidAddress = (address: string) => {
+  isValidAddress = async (address: string) => {
     const { app } = this.stores;
-    const { isMainnet, isStaging, isSelfnode } = app.environment;
-    if (isShelleyTestnet) return true;
-    const addressGroup = isIncentivizedTestnet
-      ? AddressGroup.jormungandr
-      : AddressGroup.byron;
-    const chainSettings =
-      isMainnet || isStaging ? ChainSettings.mainnet : ChainSettings.testnet;
+    const { isMainnet, isSelfnode, isStaging, isTestnet } = app.environment;
+    let expectedNetworkTag: number | null;
+    let validAddressStyles: AddressStyle[] = [];
+    if (isMainnet) {
+      expectedNetworkTag = null;
+      validAddressStyles = ['Byron', 'Icarus'];
+    } else if (isStaging) {
+      expectedNetworkTag = STAGING_MAGIC;
+      validAddressStyles = ['Byron', 'Icarus'];
+    } else if (isIncentivizedTestnet && !isShelleyTestnet) {
+      expectedNetworkTag = 0;
+      validAddressStyles = ['Jormungandr'];
+    } else if (isTestnet) {
+      expectedNetworkTag = TESTNET_MAGIC;
+      validAddressStyles = ['Byron', 'Icarus'];
+    } else if (isShelleyTestnet) {
+      expectedNetworkTag = 0;
```

Comment:
@DeeJayElly This should be what you removed in https://github.com/input-output-hk/daedalus/pull/2041/commits/316d605b9bd60f5a6d7c60123dffeb5bdc326c62